Add note to update avalon-setup submodules to Install guide

BigRoy opened the pull request:
Add install note to update the `avalon-setup` submodules using:
```
git submodule update --recursive --remote
```

This makes sure that newcomers use the latest codebase of Avalon. The newer versions have a lot of fixes, including compatibility with newer PyQt5 versions. As such this makes sure that less newcomers run into errors with PyQt5. For example, when installing with Python 3.7 then PyQt5 5.7.1 is not available, so new users would often do `pip install PyQt5` instead getting an incompatible PyQt5 for the avalon setup they pulled.

Reference:
- https://gitter.im/getavalon/Lobby?at=5decf8816a85195b9e1d23f5

BigRoy committed:
Add note to update avalon-setup submodules

Add install note to update the avalon-setup submodules

BigRoy commented:
I notice now that [updating is already described at the bottom of the page](https://getavalon.github.io/2.0/guides/#update).

1. I could change this PR to add a note right after download that describes it's recommended to do so immediately so that it is clear that the download is not directly the latest version.
2. Move *Update* to directly after download before *Command-line interface*.

With two we might avoid people accidentally reading past the note to update Avalon if they just downloaded it, but both is fine with me.

BigRoy commented:
The problem could also be alleviated maybe if we update the `avalon-setup` repository to have its submodule by default point to more recent commits. I'm not sure that actually works but I imagine as you pull it recursively that it gets the submodule commits as they were in the repository itself. As such, if we bump those versions then even without recursively pulling the latest it would already initialize to a much more recent version.

However I'm not sure if git submodules work like that.

Edit: it seems it does work like that. I'll get to it this week on updating things.

> You must run two commands: git submodule initto initialize your local configuration file, and git submodule update to fetch all the data from that project and check out the appropriate commit listed in your superproject

[Source](https://git-scm.com/book/en/v2/Git-Tools-Submodules)

BigRoy committed:
Change update message to link Update on same page and describe why.

BigRoy commented:
With https://github.com/getavalon/docs/pull/7/commits/9a0e5c56a536c1910633fcd3d9be6f22e92b8e5c I've changed it again so it shows a _Hint_ message that links to the Update at the bottom of the Guide. It also briefly states that the Download itself does _not_ load the latest version so it's clear that updating even directly after downloading might help if someone runs into issues.

Additionally I [set up a PR on `avalon-setup` so that by default it will now already be a more recent version](https://github.com/getavalon/setup/pull/7).

tokejepsen commented:
I was wondering whether we should encourage users to download and update to the latest when installing?

Since Avalon is supposed to be a safe pipeline, we should be confident that the latest versions work together. If not when we have issues to address.

davidlatwe commented:
> I was wondering whether we should encourage users to download and update to the latest when installing?

I think we should, because Avalon is an *open source project*, a reason that better than *safe* for getting latest version. :+1:

BigRoy commented:
> I think we should, because Avalon is an open source project, a reason that better than safe for getting latest version.

I'm not entirely sure I'm reading this correctly. Are you saying that we should say they are recommended to update to the latest or recommended to roll with `avalon-setup` and update when they feel they need to?

With this PR it's clear that the User didn't download the latest version with the message:

> Downloading `avalon-setup` like above does not automatically pull in the latest versions.
> See **Update** to learn how to update including all its submodules.

Would we now want to change this so that it states that we recommend updating to the latest? Or leave it as is? @tokejepsen @davidlatwe ?

tokejepsen commented:
I would say that the line `git clone https://github.com/getavalon/setup avalon-setup --recursive` should be two lines:

```
git clone https://github.com/getavalon/setup avalon-setup --recursive
git submodule update --recursive --remote
```

Then whenever users install they should get the latest version without us needing to constantly update the `avalon-setup` repository.

BigRoy commented:
I think I might agree that it's more likely to have someone run into issues because they are using an older version than someone running into "new issues" because using the latest and greatest that might introduce a bug.

I'd be fine with having the User download the latest as they get started. I know @mottosso is very wary about that and might be on the other side of that argument. @davidlatwe what would be your pick?

davidlatwe commented:
I would go for automatically downloading the latest version when getting started. 🚀 

Additionally, does git command able to do something like ignoring pre-release version when updating ?

BigRoy committed:
Always update on install, and add tip towards Update at bottom of page

BigRoy commented:
I changed it to have the guide always by default do an update to latest now. Below it I've added a tip pointing to Update at the bottom of the guide, as somehow this was previously often missed by visitors.

If there are no comments, I'll merge this on Monday.
